lib/node_modules/@stdlib/math/stats/anova1/lib/make_anova.js

```diff
@@ -0,0 +1,162 @@
+'use strict';
+
+// MODULES //
+
+var isNumberArray = require( '@stdlib/assert/is-number-array' ).primitives;
+var isTypedArrayLike = require( '@stdlib/assert/is-typed-array-like' );
+var isArray = require( '@stdlib/assert/is-array' );
+var setReadOnly = require( '@stdlib/utils/define-read-only-property' );
+var cdf = require( '@stdlib/math/base/dists/f/cdf' );
+var copy = require( '@stdlib/utils/copy' );
+var defaults = require( './defaults.json' );
+var validate = require( './validate.js' );
+var uniqueElems = require('./unique_elems.js');
+var meanTable = require('./mean_table.js');
+var prettyPrint = require( './print.js' );
+
+
+/**
+* Make the ANOVA Object.
+*
+* @param {NumericArray} x - Measured values
+* @param {Array} factor - Array of treatments
+* @param {Options} [options] - function options
+* @param {number} [options.alpha=0.05] - significance level
+* @param {boolean} [options.decision=false] - Indicate whether we reject the null hypothesis
+* @returns {Object} Returns the ANOVA Structure
```

Comment:
In JSDoc comments, we document it when the functions throw errors. Before the `@returns` annotation, we should add:

```
* @throws {TypeError} options argument must be an object
* @throws {TypeError} must provide valid options
* @throws {TypeError} `x` must be a numeric array
* @throws {TypeError} `factor` must be an array
* @throws {RangeError} `factor` must have at least two unique elements
* @throws {RangeError} length of `x` must be greater than or equal to 2
* @throws {RangeError} `x` and `factor` must have the same length
```

Reply:
Resolved. Must the text match the error messages generated in the `throw new`?

Reply:
No, not necessarily. There should, however, be some level of agreement such that people reading the code do not see conflicting messages about an error's intent.